src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public void setOperation(String operation) {
+        Optional<Operation> matchingOperation = Arrays.stream(Operation.values())
+                .filter(val -> val.getCode().equals(operation))
+                .findFirst();
+
+        if (matchingOperation.isPresent()) {
+            this.operation = matchingOperation.get();
+        } else {
+            throw new IllegalArgumentException("Invalid operation code: " + operation);
+        }
```

Comment:
It's good to see you've provided a way to set the operation based on a string code. However, consider encapsulating this logic within the `Operation` enum itself, perhaps with a static method that performs this lookup. This would adhere more closely to the single responsibility principle by keeping the operation lookup logic within the `Operation` enum.